Aero_Sub_DAQ/Air_pressure_sensor_code/air_pressure_code_v2.ino

```diff
@@ -0,0 +1,155 @@
+/*
+* Air Pressure Sensor
+* Authors:        Allison Pham         Embedded Firmware Team
+*                 Srinidhi Malladi
+*                 William Everson
+* 
+* Email:          willreverson@gmail.com
+* 
+* (c) 2022 Dallas Formula Racing - Formula SAE
+* GPL-3.0 License
+*/
+
+#include <Wire.h>
+#include <SPI.h>
+#include <SD.h>
+
+#define I2C_ADDRESS 0x29
+
+const int kSwitchPin = 7, kChipSelect = 10;
+const float kTwo24 = 16777216.00;
+uint8_t cmd[8], StatusByte, cmdbyte;
+uint32_t outb[12];
+int file_name_num = 1, count;
+float time;
+String file_name;
+File dataFile;
+bool switchUsed;
+
+float temperatureTransferFunction(uint32_t outb[]) {
+  uint32_t Tmp = 0;
+  float fTemp;
+  // --- convert Temperature to degrees C:
+  Tmp = (outb[4] << 8) + outb[5];
+  fTemp = (((float)Tmp*125.0)/kTwo24) - 40.0;
+  Serial.print("Temperature: ");
+  Serial.println(fTemp);
+  return abs(fTemp);
+}
+
+float pressureTransferFunction(uint32_t outb[]) {
+  uint32_t Prs = 0;
+  float fPress, transf;
+
+  // --- convert Pressure to %Full Scale Span ( +/- 100%)
+  Prs = (outb[1] << 16) + (outb[2] <<8) + outb[3];

```

Comment:
Recall that we did a brute-force method of allocating 32 bits for every element of the byte array. We will need to optimize this by doing compile-time casting into different data sizes **before** the shifting is performed.

We can leave this comment to be unresolved for now and focus on everything else. This can be addressed later.